aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/SubscriptionProcessor.java

```diff
@@ -119,6 +122,8 @@ synchronized void startSubscriptions() {
             return;
         }
         if (subscriptionsStarted) {
+            Amplify.Hub.publish(HubChannel.DATASTORE,
```

Comment:
We have full liberty to add a new `publish(...)` method, if you would like to, as well!

Maybe like:
```java
<E extends Enum<E>> void publish(HubChannel channel, Enum<E> eventName) {
    publish(channel, HubEvent.create(eventName));
}
```
A small optimization to reduce some boiler plate. I've already baked in some `Enum<E>` type mumbo-jumbo elsewhere around the Hub APIs.

Reply:
Oh...I will do that then. I think that will go a long way in cleaning some of this stuff up.

Reply:
Well, turns out it's not as much on the publishing side as it is on the subscriber side. The `HubSubscriber` interface currently doesn't have a method that takes in a generic type. Modifying the existing one would probably make this a breaking change. Adding a new method to that interface would then force us to change every calling site from lambda-style to an anonymous object declaration. Neither of those sounded like a good plan.

Instead, I went with adding a factory method in `HubSubscriber` which makes the calling code a little cleaner.